Modules/mathmodule.c

```diff
@@ -1207,10 +1194,11 @@ math_floor(PyObject *module, PyObject *number)
         }
         if (PyErr_Occurred())
             return NULL;
-        x = PyFloat_AsDouble(number);
-        if (x == -1.0 && PyErr_Occurred())
-            return NULL;
     }
+    double x = PyFloat_AsDouble(number);
```

Comment:
I missed this on my first review, but PyFloat_AsDouble is not the same as PyFloat_AS_DOUBLE, since the latter is faster. Maybe we should make math_ceil match the math_floor logic (best done in separate PR)?

Reply:
Yup, this was an optimisation added in https://github.com/python/cpython/pull/21072. I'll open a PR making math_ceil match the logic here

Reply:
https://github.com/python/cpython/pull/108801

Reply:
Thanks, I've reverted the math_floor() change and added new tests to cover L1239.  (And similar tests for ceil() to help you bring coverage back in #108801.)